src/js/select2/data/ajax.js

```diff
@@ -88,7 +88,7 @@ define([
       self._request = $request;
     }
 
-    if (this.ajaxOptions.delay && params.term !== '') {
+    if (this.ajaxOptions.delay && params.term !== undefined) {
```

Comment:
I believe the goal of checking for `''` was to handle the case when someone completely empties the search box (deletes everything). Assuming that _is_ when this case was previously triggered, it's something we're going to need to maintain.